File: src/main/java/cryorbiter/mojito_spb/model/Facture.java
package cryorbiter.mojito_spb.model;

import java.util.Date;

import jakarta.persistence.*;
import cryorbiter.mojito_spb.enumerations.StatutFacture;
import cryorbiter.mojito_spb.enumerations.TypeDocument;

@Entity
@Table(name = "factures")
public class Facture extends Document {

    public static final long serialVersionUID = 13L;

    @Enumerated(EnumType.STRING)
    @Column(name = "statut_facture", nullable = false)
    private StatutFacture statutFacture;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "commande_id")
    private Commande commande;

    public Facture() {}

    public Facture(long id, Date date, float poids, float tva, TypeDocument type, String libelle, Double prixKilo,
            StatutFacture statutFacture, Commande commande) {
        super(id, date, poids, tva, type, libelle, prixKilo);
        this.statutFacture = statutFacture;
        this.commande = commande;
    }

    public StatutFacture getStatutFacture() {
        return statutFacture;
    }

    public void setStatutFacture(StatutFacture statutFacture) {
        this.statutFacture = statutFacture;
    }

    public Commande getCommande() {
        return commande;
    }

    public void setCommande(Commande commande) {
        this.commande = commande;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    @Override
    public String toString() {
        return "Facture{" +
                "id=" + getId() +
                ", date=" + getDate() +
                ", poids=" + getPoids() +
                ", tva=" + getTva() +
                ", type=" + getType() +
                ", libelle='" + getLibelle() + '\'' +
                ", prixKilo=" + getPrixKilo() +
                ", commentaire='" + getCommentaire() + '\'' +
                ", nomDocument='" + getNomDocument() + '\'' +
                ", client=" + getClient() +
                '}';

    }
}

File: src/main/resources/static/js/facture.js
document.addEventListener('DOMContentLoaded', () => {
    const selectCommande = document.querySelector('#commandeId');

    if (!selectCommande) return;

    selectCommande.addEventListener('change', async () => {
        const commandeId = selectCommande.value;
        if (!commandeId) return;

        try {
            const response = await fetch(`/facture/commande/${commandeId}`);
            if (!response.ok) throw new Error('Commande introuvable');

            const commande = await response.json();

            // Remplissage dynamique
            document.querySelector('[name="nomDocument"]').value = commande.nomDocument || '';
            document.querySelector('[name="libelle"]').value = commande.libelle || '';
            document.querySelector('[name="date"]').value = commande.date || '';
            document.querySelector('[name="poids"]').value = commande.poids || '';
            document.querySelector('[name="tva"]').value = commande.tva || '';
            document.querySelector('[name="prixKilo"]').value = commande.prixKilo || '';
            document.querySelector('[name="type"]').value = commande.type || '';
            document.querySelector('[name="statutFacture"]').value = commande.statutCommande || '';
            document.querySelector('[name="commentaire"]').value = commande.commentaire || '';

            // Remplir les IDs cachés
            document.getElementById("commandeId").value = commande.id;
            if (commande.client) document.getElementById("clientId").value = commande.client.id;

        } catch (error) {
            console.error(error);
            alert('Impossible de charger les informations de la commande');
        }
    });
});
